### PR TITLE
Update Tuist to build with Xcode 14.2

### PR DESCRIPTION
**Pull request by thedavidharris**
### Short description 📝

Moves Tuist to compile on CI and build with Xcode 14.1. This unblocks a PR related to multi-platform build targets and better aligns with current development versions.

### Contributor checklist ✅

- [x] The code has been linted using run `./fourier lint tuist --fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] Contributors have checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/ProjectDescription/Config.swift [hunk 1]
```diff
@@ -21,8 +21,8 @@
 /// import ProjectDescription
 ///
 /// let config = Config(
-///     compatibleXcodeVersions: ["10.3"],
-///     swiftVersion: "5.4.0"
+///     compatibleXcodeVersions: ["14.3"],
```
Comment (thedavidharris) [inline Sources/ProjectDescription/Config.swift:24]: Just to trigger CI

File: .xcode-version-libraries [hunk 2]
```diff
@@ -1 +1 @@
-13.4.1
\ No newline at end of file
+14.1
```
Comment (danyf90) [inline .xcode-version-libraries:1]: I think this should remain the old one to not impact the users.

**Review by danyf90 — CHANGES_REQUESTED** [review]
I think you also need to update the CI jobs to use the new images or Xcode builds, isn't it?

**Comment by kwridan** [comment]
heads up there's a high chance we'll run into this issue https://github.com/apple/swift-package-manager/issues/5767 😞

**Comment by thedavidharris** [comment]
I'm not going to get a chance to follow this up in the short-term, so closing this for someone else to pick up.

https://github.com/tuist/tuist/pull/5005 depends on a similar change.

**Comment by danyf90** [comment]
I'll try to fix this 🤞

**Review by danyf90 — APPROVED** [review]

**Pull request closed without merging**